src/cpp/utils/eClock.cpp

```diff
@@ -16,126 +16,72 @@
  * @file eClock.cpp
  *
  */
-#include <cmath>
-#include <iostream>
+
 #include <fastrtps/utils/eClock.h>
 
+#include <chrono>
+#include <thread>
+
+using namespace std::chrono;
+
 namespace eprosima {
 namespace fastrtps {
 
-//FIXME: UTC SECONDS AUTOMATICALLY
 eClock::eClock()
-    : m_seconds_from_1900_to_1970(0)
-    , m_utc_seconds_diff(0)
 {
-#if defined(_WIN32)
-    QueryPerformanceFrequency(&freq);
-#endif
 }
 
-eClock::~eClock() {
-
-}
-
-
-#if defined(_WIN32)
-#include <cstdint>
-
-bool eClock::setTimeNow(fastrtps::Time_t* tnow)
+eClock::~eClock()
 {
-
-    GetSystemTimeAsFileTime(&ft);
-    ftlong = ft.dwHighDateTime;
-    ftlong <<=32;
-    ftlong |= ft.dwLowDateTime;
-    ftlong /=10;
-    ftlong -= 11644473600000000ULL;
-
-    //std::cout << "ftlong: " << ftlong << std::endl;
-	//std::cout << "sec from 1900 " << m_seconds_from_1900_to_1970<<std::endl;
-	tnow->seconds = (int32_t)((long)(ftlong/1000000UL)+(long)m_seconds_from_1900_to_1970+(long)m_utc_seconds_diff);
-	//std::cout << "seconds: " << tnow->seconds << " seconds " << std::endl;
-	tnow->nanosec = (uint32_t)((long)(ftlong%1000000UL));
-	return true;
 }
 
-bool eClock::setTimeNow(rtps::Time_t* tnow)
+static void current_time_since_unix_epoch(
```

Comment:
Duplicated function in `Time_t.cpp`. It's very likely we can omit this version and use the new `Time_t` methods to get the same functionality.